### ISSUE TITLE
AWS Lambda updateFunctionCode() method does not work in PHP

### ISSUE THREAD

**Issue by pako-pl** [aws/aws-sdk-php#580]
````
$lambdaClient->updateFunctionCode(array(
      'FunctionName' => 'TestFunction',
      'ZipFile' => base64_encode(file_get_contents('test.zip')),
));
````

The 'ZipFile' value seems to be skipped and the request has no contents:

````
# Request:
PUT /2015-03-31/functions/TestFunction/versions/HEAD/code HTTP/1.1
Host: lambda.eu-west-1.amazonaws.com
User-Agent: aws-sdk-php2/2.8.4 Guzzle/3.9.3 curl/7.26.0 PHP/5.6.9
x-amz-date: 20150518T140640Z
Authorization: AWS4-HMAC-SHA256 Credential=XXXXXXXXXXXXXXXXXX/eu-west-1/lambda/aws4_request, SignedHeaders=host;x-amz-date, Signature=XXXXXXXXXXXXXXXXXXXXXXXXXX
Content-Length: 0


# Response:
HTTP/1.1 400 Bad Request
Content-Type: application/json
Date: Mon, 18 May 2015 14:06:41 GMT
x-amzn-ErrorType: InvalidParameterValueException:http://internal.amazon.com/coral/com.amazonaws.awsgirapi/
x-amzn-RequestId: 1b7c3081-fd67-11e4-a30d-476b25cb4f44
Content-Length: 70
Connection: keep-alive

{"Type":"User","message":"Please provide a source for function code."}
````

Uploading the same file through the AWS Console works fine.

**Comment by mtdowling** [comment]
Looks like this is an issue similar to one we've had before with Lambda in that the "ZipFile" needs a location of "body". We'll take a look at our service description generator to see why this is happening to fix this at the source.

**Comment by pako-pl** [comment]
Any suggestions as to how to work around this issue until the bug gets fixed? Right now we have to manually deploy our changes to Lambda functions.

**Comment by halkyon** [comment]
According to https://github.com/aws/aws-sdk-php/releases/tag/2.8.29 it was fixed in 2.8.29, but I'm running this version and still getting the same "Please provide a source for function code." result even though I'm passing a base64-encoded string to the `ZipFile` argument.

**Comment by jeskew** [comment]
@halkyon I was able to reproduce the error you saw and fixed it with #980. The fix will be included in the next 2.8.x release.

**Comment by halkyon** [comment]
@jeskew awesome! Thanks for that.
